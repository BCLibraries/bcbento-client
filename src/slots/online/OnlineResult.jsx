import React from 'react';
import VideoResult from "../video/VideoResult";
import {PrimoRecordLink} from "../PrimoRecordLink";
import HathiTrustLink from "../books/HathiTrustLink";
import AvailabilityContainer from "../availability/AvailabilityContainer";
import CoverImage from "../CoverImage";
import GetItLink from "../GetItLink";
import {getDisplayCreator} from "../getDisplayCreator";

function OnlineResult({item}) {
    if (item.isElectronic && item.type === 'Video') {
        return <VideoResult item={item} inCatalogResult={true}/>;
    }

    const link = PrimoRecordLink(item);

    return <li className="catalog-result-item media">

        <div className="media-body">
            <h3 className="catalog-result-item__media-heading media-heading">
                <a href={link} className="catalog-result-item__title">
                    {item.title}
                </a>
            </h3>

            <div className="catalog-result-item__creator">{getDisplayCreator(item)}</div>

            <div className="catalog-result-item__publisher">
                {item.publisher} {item.date}
            </div>

            <div className="catalog-result-item__type">{item.type}</div>

            {item.isElectronic && <GetItLink url={link}/>}

            {item.isPhysical && <AvailabilityContainer item={item}/>}

            {item.hathitrustUrl && <HathiTrustLink url={item.hathitrustUrl}/>}

        </div>

        {item.coverImages.length > 1 && <CoverImage itemUrl={link} imageUrl={item.coverImages[0].url}/>}
    </li>
}

export default OnlineResult;