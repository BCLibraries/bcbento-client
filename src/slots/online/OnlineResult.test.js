import renderer from 'react-test-renderer';
import OnlineResult from './OnlineResult.jsx';

it('renders correctly', () => {
    const item = {
        "id": "ALMA-BC51420267560001021",
        "title": "English",
        "date": "2008",
        "type": "Video",
        "creator": null,
        "contributors": [
            "Available Light Productions"
        ],
        "coverImages": [
            {
                "url": "no_cover",
            }
        ],
        "available": null,
        "hathitrustUrl": null,
        "isPhysical": false,
        "isElectronic": true,
        "screenCap": null,
        "mms": "99131628470001021",
        "holdings": [],
    };

    const component = renderer.create(
        <OnlineResult item={item}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
