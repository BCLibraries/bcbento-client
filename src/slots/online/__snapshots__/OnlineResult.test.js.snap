// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="catalog-result-item"
>
  <div
    className="online-video"
  >
    <a
      href="https://mlib.bc.edu/reserves-api/items/99131628470001021"
    >
      <div
        className="online_video__thumb"
      >
        <img
          alt="Watch English online"
          className="online-video-thumb__blank_screen"
          onError={[Function]}
          src="https://library.bc.edu/images/blank-screen.svg"
        />
      </div>
    </a>
    <div>
      <div
        className="catalog-result-item__type"
      >
        Video
      </div>
      <h3
        className="online-video__media-heading media-heading"
      >
        <a
          className="online-video__title"
          href="https://bc-primo.hosted.exlibrisgroup.com/primo-explore/fulldisplay?docid=ALMA-BC51420267560001021&context=L&tab=bcl_only&search_scope=bcl&vid=bclib_new&lang=en_US"
        >
          English
        </a>
      </h3>
    </div>
    2008
    <div
      className="online-video__creator"
    >
      Available Light Productions
    </div>
    <div
      className="online-video__watch-link"
    >
      <a
        href="https://mlib.bc.edu/reserves-api/items/99131628470001021"
      >
        Watch online
      </a>
    </div>
  </div>
</li>
`;
